catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/CatalogFrameworkImpl.java

```diff
@@ -2407,6 +2407,8 @@ protected ResourceResponse getResource(ResourceRequest resourceRequest, boolean
                             e);
                 }
             }
+
+            resourceResponse.getProperties().put(Constants.METACARD_PROPERTY, metacard);
```

Comment:
Generally, minimizing side effects results in a better API.  I think it's pretty confusing for a method signature to say, "I'll accept a request object and return a response object and a portion of what you need will be returned via the request."  If there's a use-case to have it on the request then we should put it on both objects. My current use case only needs it on the response.